week1/README.md

```diff
@@ -0,0 +1,220 @@
+
+The navigation bar in the image is made up of 7 instances of the same component, let's call it `NavbarItem`. We make sure we know what the content is of each instance: in this case the names for each page in the website. And then we just pass those values to each instance.
+
+Essentially, a component is a custom HTML element that we define ourselves.
+
+Learn more by going through the following resources:
+
+- [ReactJS Component LifeCycle](https://www.youtube.com/watch?v=3EbYJrAOpUs)
+- [Components and Props](https://reactjs.org/docs/components-and-props.html)
+
+### State & Props
+
+What makes React `dynamic` is the concept of `state`. Simply put, state is the data that moves from component to component. It changes based on the input from the user. Let's say a user logs into their Facebook account. In React terms this is what will happen: A user types in their account details into the login form. This data will be placed into the application state, in real-time.
+
+This means: The other parts of the application knows immediately about the content you've just written! So if we want to use those exact same details to show in the next screen, we can do so without having to send a request to the server. All of this user-specific data stays inside the client.
+
+The other side of `state` is the how we pass it down to our other components. When this happens we call it `props`. It's short for `property`, similar to the properties in HTML elements.
+
+`props` are passed down using a self-defined property name. Take a look at the following example:
+
+```js
+const Parent = () => {
+  return <Child thisIsaProp={this.state.randomPropName} />;
+};
+```
+
+The `thisIsaProp` name is something I've created myself, it has no special meaning whatsoever. However, while that is the case it's still important to make any name you give it semantic: it should be meaningful and indicate its purpose. For example:
+
+```js
+const Parent = () => {
+  return <Child user={this.state.user} />;
+};
+```
+
+In the above example, the `prop` has a name that describes its contents. From it we can infer that it's probably a user object, with user-specific information.
+
+Learn more about state and props here:
+
+- [State and Lifecycle](https://reactjs.org/docs/state-and-lifecycle.html)
+- [ReactJS State Tutorial](https://www.youtube.com/watch?v=DPdc5Z-Tf4U)
+- [ReactJS Props](https://www.youtube.com/watch?v=eXQRhVUMCP8)
+
+### Lifecycle
+
+If you only build static websites, you are used to pages rendering instantly after they are send to the client. Whether it comes from the server, or you directly open an HTML file by dragging it into your browser, it displays immediately.
+
+This goes differently in React. Every component has a `lifecycle`. Like a human being, you are born, grow up to do certain activities, and ultimately, pass away. The same thing applies here. Only in the case of React components this happens in 3 main phases:
+
+1. Mounting
+2. Updating
+3. Unmounting
```

Comment:
maybe emphasising on lifecycles could be confusing.
the hook model is more expressive since it is state synchronisation rather than lifecycles making it hard to relate and difficulty understanding lifecycle -> hook mental model

This was honestly one thing i struggled understanding the hook model at first look, till this tweet cleared it for me.
https://twitter.com/ryanflorence/status/1125041041063665666